app/assets/javascripts/discourse/app/controllers/reject-reason-reviewable.js

```diff
@@ -0,0 +1,16 @@
+import Controller from "@ember/controller";
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+
+export default Controller.extend(ModalFunctionality, {
+  onShow() {
+    this.setProperties({ rejectReason: "", sendEmail: false });
+  },
+
+  actions: {
+    perform() {
+      this.model.set("rejectReason", this.rejectReason);
```

Comment:
You're not setting `sendEmail` here, because it seems bound in the template to `model.sendEmail` not the local variable. In that case, if a user checks off `sendEmail` then closes the modal, it'll still be true when they submit the reviewable. It needs to be reset, and preferably to use the component's `sendEmail`